src/core/modules/SolarSystem.cpp

```diff
@@ -1621,22 +1633,52 @@ void SolarSystem::fillEphemerisDates()
 	}
 }
 
+// a wrapper for getting an IAU designations of minor bodies
+QString SolarSystem::getIAUDesignation(PlanetP minorBody) const
+{
+	QString iau;
+	if (minorBody->getPlanetType()==Planet::isComet)
```

Comment:
This would invite implementing a virtual method 
`Planet::getIAUDesignation() const {return englishName;}`

which is overridden in MinorPlanet and Comet classes. Then this method is trivially:

```
QString SolarSystem::getIAUDesignation(PlanetP p) const
{ 
    return p->getIAUDesignation();
}
```

Reply:
It could also simplify the methods below.